pytorch_lightning/accelerators/tpu.py

```diff
@@ -59,3 +59,18 @@ def _move_optimizer_state(self, device: Optional[torch.device] = None) -> None:
         for opt in self.optimizers:
             for p, v in opt.state.items():
                 opt.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, self.root_device)
+
+    def get_device_stats(self, device: Optional[torch.device] = None) -> Dict[str, Any]:
+        """Gets stats for the given TPU device."""
+        device_stats = {}
+        memory_info = xm.get_memory_info(device)
+
+        free_memory = memory_info["kb_free"]
+        peak_memory = memory_info["kb_total"] - free_memory
+
+        free_memory = self.training_type_plugin.reduce(free_memory) * 0.001
+        peak_memory = self.training_type_plugin.reduce(peak_memory) * 0.001
```

Comment:
@kaushikb11 n00b question: why do we need the reduce here?

Reply:
I guess this is used to compute the mean across the different tpu_cores.

Reply:
I agree with @tchaton - the accelerator ought to report per-device stats here (this is what the docstring says too). So I don't think we should be doing any reductions using the training type. That can be let to the calback/some other agent for how they want to aggregate these stats

@kaushikb11 what do you think?

Reply:
sg I removed the reduce